### PR TITLE
Created a service and move the main functionality (requests sending and results processing) into it.

### PR DESCRIPTION
**Pull request by Flame787898**
https://redmine.devbranch.work/issues/5840

### CODE REVIEW

File: web/modules/custom/exchange_rates/exchange_rates.module [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+/**
+ * @file
+ * Hook theme.
```
Comment (Y0r) [inline web/modules/custom/exchange_rates/exchange_rates.module:5]: bad

File: web/modules/custom/exchange_rates/exchange_rates.services.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+services:
+  exchange_rates.api_connector:
+    class: Drupal\exchange_rates\ExchangeAPIConnector
+    arguments: ['@http_client','@config.factory']
```
Comment (Y0r) [inline web/modules/custom/exchange_rates/exchange_rates.services.yml:4]: space between arguments

File: web/modules/custom/exchange_rates/src/ExchangeAPIConnector.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  public function getExchangeConfig() {
+    $config_form = $this->configForm->get('exchange_rates.settings');
+    return $config_form;
+  }
+
+  /**
+   * @return \Psr\Http\Message\StreamInterface|void
+   *   Return data from request
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   */
```
Comment (Y0r) [inline web/modules/custom/exchange_rates/src/ExchangeAPIConnector.php:48]: ?

File: web/modules/custom/exchange_rates/src/ExchangeAPIConnector.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  public function __construct(ClientInterface $client, ConfigFactory $config) {
+    $this->httpClient = $client;
+    $this->configForm = $config;
+  }
+
+  /**
+   * @return \Drupal\Core\Config\Config|\Drupal\Core\Config\ImmutableConfig
+   *   Rerun config form seating
+   */
```
Comment (Y0r) [inline web/modules/custom/exchange_rates/src/ExchangeAPIConnector.php:38]: ?

**Review by Y0r — CHANGES_REQUESTED** [review]
Solve problems first

File: web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+class ExchangeRatesBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * @var \Drupal\exchange_rates\ExchangeAPIConnector
+   *   In this variable are stored service methods.
+   */
+  protected $exchangeAPIConnector;
+
+  /**
+   * @param array $configuration
```
Comment (aidua) [inline web/modules/custom/exchange_rates/src/Plugin/Block/ExchangeRatesBlock.php:27]: fix coding standards here and below.

**Pull request merged**